lockfile/src/parsers/pypi.rs

```diff
@@ -52,10 +53,26 @@ fn line(input: &str) -> IResult<&str, &str> {
         line = "";
     }
 
+    // Ignore index config options.
+    //
+    // Since `ThirdPartyVersion` only allows a single registry, we prefer recording
+    // only the primary one.
+    if let Some(index_url) = line.strip_prefix("--index-url") {
+        *registry = Some(index_url.trim());
+        line = "";
+    }
+    if let Some(extra_index_url) = line.strip_prefix("--extra-index-url") {
+        // Only use extra index URL if no other URL has been specified.
```

Comment:
The primary registry when `--index-url` is not provided but `--extra-index-url` is provided will still be `https://pypi.org/simple`

Reply:
The "primary registry" just means the one that was specified with `--index-url`. It is not special in any other way.

So when `--extra-index-url` is specified, we have no way to know which index will be used... We must make an assumption. I thought it would be safer to assume third party.